group-actors/src/main/java/org/sunbird/util/ActivitySearchRequestGenerator.java

```diff
@@ -0,0 +1,136 @@
+      String authToken = System.getenv(headers.get(JsonKey.AUTHORIZATION));
+      if (!StringUtils.isNotBlank(baseUrl)) {
+        baseUrl =
+            PropertiesCache.getInstance()
+                .getProperty((String) activityConfig.get(JsonKey.BASE_URL));
+      }
+      if (!StringUtils.isNotBlank(apiUrl)) {
+        apiUrl =
+            PropertiesCache.getInstance().getProperty((String) activityConfig.get(JsonKey.API_URL));
+      }
+      if (!StringUtils.isNotBlank(authToken)) {
+        authToken = PropertiesCache.getInstance().getProperty(headers.get(JsonKey.AUTHORIZATION));
+      }
+      headers.put(JsonKey.AUTHORIZATION, JsonKey.BEARER + authToken);
+      for (String activityType : (List<String>) activityConfig.get("type")) {
+        String searchApiUrl = baseUrl + apiUrl;
+        activityServiceSearchUrlMap.put(activityType, searchApiUrl);
+        activityServiceSearchRequestMap.put(
+            searchApiUrl, (Map<String, Object>) activityConfig.get(JsonKey.REQUEST));
+        activityServiceResponseFieldMap.put(
+            searchApiUrl, (String) activityConfig.get(JsonKey.RESPONSE));
+        activityServiceHeaderMap.put(searchApiUrl, headers);
+      }
+    }
+  }
+
+  public static List<ActivitySearchRequestConfig> generateActivitySearchRequest(
+      List<Map<String, Object>> activities) {
+    Map<String, List<String>> idListMap = new HashMap<>();
+    for (Map<String, Object> activity : activities) {
+      String apiUrl = (String) activityServiceSearchUrlMap.get(activity.get(JsonKey.TYPE));
+      if (idListMap.containsKey(apiUrl)) {
+        List<String> identifiers = idListMap.get(apiUrl);
+        identifiers.add((String) activity.get(JsonKey.ID));
+      } else {
+        List<String> identifiers = new ArrayList<>();
+        identifiers.add((String) activity.get(JsonKey.ID));
+        idListMap.put(apiUrl, identifiers);
+      }
+    }
+    List<ActivitySearchRequestConfig> activitySearchRequestConfigs =
+        createMultiSearchActivityRequest(idListMap);
+
+    return activitySearchRequestConfigs;
+  }
+
+  private static List<ActivitySearchRequestConfig> createMultiSearchActivityRequest(
+      Map<String, List<String>> idListMap) {
+    List<ActivitySearchRequestConfig> activitySearchRequestConfigs = new ArrayList<>();
+
+    for (Map.Entry<String, List<String>> idList : idListMap.entrySet()) {
+      ActivitySearchRequestConfig activitySearchRequestConfig = new ActivitySearchRequestConfig();
+      SearchRequest searchRequest = new SearchRequest();
+      String apiUrl = idList.getKey();
+      List<String> identifiers = idList.getValue();
+      Map<String, Object> requestPayload = new HashMap<>();
+      Map<String, Object> filters = new HashMap<>();
+      Map<String, Object> requestConfig = activityServiceSearchRequestMap.get(apiUrl);
+      filters.put((String) requestConfig.get(JsonKey.ID), identifiers);
+      requestPayload.put(JsonKey.FILTERS, filters);
```

Comment:
@sknirmalkar89 - there is an assumption that APIs for all activity types will accept "filters" in their  request body. It is not a valid assumption.